### PR TITLE
FUSE Support and build.sh cleanup

### PR DESCRIPTION
**Pull request by Ranbato**
@sdub76 please take a look and verify that it works for you.

### CODE REVIEW

**Comment by Sdub76** [comment]
I tried it and I can execute the mount command from the GUI and it says it successfuly mounted, but the folder is inaccessible from the docker container (docker exec -it vorta /bin/bash).  Nor can I mount it to a folder that's mapped into the docker image.  (Not sure if that could possibly work anyway).

`ls: ./vorta_mount: Permission denied
`
I have privleged mode turned on, and all of the extra options pased.

```
docker run -d --name='vorta' --net='bridge' --privileged=true -e TZ="America/New_York" -e HOST_OS="Unraid" -e 'TZ'='America/New_York' -p '5811:5800/tcp' -v '/mnt/user/appdata/vorta-docker':'/config':'rw' -v '/mnt/user/':'/data':'ro' -v '/mnt/disks/borg_backup/':'/destination':'rw' -v '/mnt/user/downloads/vorta_mount':'/mnt/vorta_mount':'rw' --device='/dev/fuse' --cap-add SYS_ADMIN --security-opt apparmor=unconfined 'ghcr.io/borgbase/vorta-docker:local'
9d4463ff3974f1c1429d4e3caafc655298adab7b884c195390937280c4ebbcd4
```

**Comment by Ranbato** [comment]
Weird behavior that I can't explain yet.
Exec'ing into the container and running `borg mount /my/archive /destination/temp` works and I can explore the archive just fine.  Doing it from the UI says the mount succeeds, but then the mounted directory becomes inaccessible with a **permissions denied** error.

TL;DR command line works, UI mounts successfully but something makes the mount unavailable

**Comment by Sdub76** [comment]
I conirmed that I can manually mount the archive from the command line, but I get

`docker: Error response from daemon: unable to find user app: no matching entries in passwd file.`

if I add --user app to the docker run command.  It's odd because you'd think the default root user would be able to access anything.  Is it possible to manipulate the UID/GID to 0/0 to run Vorta as root within the docker?

**Comment by Sdub76** [comment]
It's also worth adding that I only required the 

--device='/dev/fuse' --cap-add SYS_ADMIN 

options for the command-line mount to work properly in my tested environment (Unraid/Docker version 20.10.5)

**Comment by m3nu** [comment]
> TL;DR command line works, UI mounts successfully but something makes the mount unavailable

If that's the case, worth checking the command Vorta uses. Is Vorta currently running in the foreground, so we get the logs via the container?

**Comment by Ranbato** [comment]
This container is specifically set up to follow Docker best practices and not allow Vorta to run as root.  Therefore the mount is getting created only for the **app** user and not for root.  Since `docker exec` defaults to running as root, you don't have access to the mount.
The original command I gave was a mistake as it isn't `docker run` that requires it, it is the exec like so ` docker exec -it --user app vorta sh`

Try that.

**Comment by Sdub76** [comment]
Yes, I can confirm that this works.  Thanks.

Is there any way this could be done in a way that exposes the mount to the underlying OS?  Such that I pass in a /mnt/borg folder to the docker container, which mounts the archive and allows the host OS to browse its contents?

I'm comfortable doing all of this from the Borg command line, but my goal in getting this Vorta container working is for less tech-saavy family members.  If they could mount an archive to the default location and just browse to it via samba that would be so much easier for them to understand.

**Comment by Sdub76** [comment]
I would agree that xterm doesn’t make sense. It’s easy enough to docker
exec into a container and many docker front ends provide this by default
(docker desktop, Unraid, etc.)

**Comment by Sdub76** [comment]
I would agree that adding xterm doesn’t make sense. It’s easy enough to docker exec into a container and many docker front ends provide this by default (docker desktop, Unraid, etc.)
